### PR TITLE
type::TypeId using enum class

### PR DESCRIPTION
**Pull request by saatviks**
Addressing #851

### CODE REVIEW

**Comment by apavlo** [comment]
This is great! I am surprised that it didn't require more changes.

**Comment by saatviks** [comment]
I was testing everything on the Release build. But it seems like I should have been using the Debug build.
I guess I should build with 'cmake -DCMAKE_BUILD_TYPE=Debug ..' and ensure that `make check` passes on this.

File: src/traffic_cop/traffic_cop.cpp [hunk 1]
```diff
@@ -493,7 +493,8 @@ FieldInfo TrafficCop::GetColumnFieldForValueType(std::string column_name,
     default: {
       // Type not Identified
       LOG_ERROR("Unrecognized field type '%s' [%d] for field '%s'",
-                TypeIdToString(column_type).c_str(), column_type,
+                TypeIdToString(column_type).c_str(),
+                static_cast<int>(column_type),
```
[the remark this answers is not in the document]
Comment (saatviks) [reply src/traffic_cop/traffic_cop.cpp]: I'm not sure what you mean here? Which part is redundant?

[the remark this answers is not in the document]
Comment (seojungmin) [reply src/traffic_cop/traffic_cop.cpp]: @saatviks It was not very clear.  We do not need both the `string` version and the `int` version of `column_type` in the error printout.  We could remove the `int` version: something like `LOG_ERROR("Unrecognized field type '%s' for field '%s'", TypeIdToString(column_type).c_str(), column_name.c_str());`. I know that there are these kinds of unnecessary code in other places, but we need to at least try to make the code compact as possible.

**Pull request merged**